lib/createTopLevelExpect.js

```diff
@@ -1286,13 +1286,14 @@ expectPrototype.lookupAssertionRule = function(
 };
 
 expectPrototype._assertTopLevelExpect = function() {
-  if (this !== this._topLevelExpect) {
+  // https://github.com/unexpectedjs/unexpected/issues/631
+  if (this.flags) {
```

Comment:
It is really frustrating that we have to do this but I agree that `esm` is widely used enough we probably need to.

That said, the new check feels really loose.. is there a chance we should set a new boolean on all expects like _isTopLevelExpect and check this as a boolean flag in all these places? I'm a little nervous about checking something like this.flags since it is just a side effect and who knows if we change it later.

Reply:
Now that the top level expect is actually in the prototype chain of the wrapped expect, `expect._isTopLevelExpect` would always come out as true, so that wouldn't work. It would have to be an `_isWrappedExpect` property that we only set on wrapped expects. I didn't feel like introducing another property there as we've spent a long time getting the number of properties on that thing down.

Maybe a better way to ensure that it keeps working could be to add `esm` to the test suite.

Reply:
Yeah I know it was considerable work to get rid of them and I'm loathed to re-introduce it too. If we put the GitHub issue on top of the `!this.flags` case a little further down I can live with that since it'll have been documented :) Would that be ok?

Reply:
You've got yourself a deal, fixed in d7a6a645